src/reducers/players.js

```diff
@@ -60,6 +61,18 @@ const players = (state = initialState, action) => {
         }),
       });
     }
+    case GAIN_MANA: {
+      let newMana = action.mana + state[action.playerId].mana;
+      if (newMana > 10) {
+        newMana = 10;
+      }
+
+      return Object.assign({}, state, {
```

Comment:
Use lodash's `merge` instead of 2 nested `Object.assigns()`. I think `merge` is always preferable except when we are removing object properties or removing array values--then you have to use nested `Object.assigns()`